### ISSUE TITLE
Drawing tools, Drawing selection

### ISSUE THREAD

**Issue by OstapchelaEst** [amcharts/amcharts5#1460]
Good afternoon. After the recent release [(5.9.1](https://github.com/amcharts/amcharts5/releases/tag/5.9.1)) a new `drawingSelectionEnabled` parameter has been added to stcokChart, which by default is false and prohibits moving drawings.  

My goal is to make it so that drawings can always be moved.  To do this, I pass this parameter with the value `true`.

```
const DRAWING_CONTROLL = am5stock.DrawingControl.new(root, {
  stockChart,
  strokeColor: am5.color(0xff0000),
  drawingSelectionEnabled: true
})
```

After starting the application they are indeed all moved, but if i add another drawing this parameter becomes false and the drawings become immobile. To add a new drawing I use my own button with the following code

```
document.getElementById('draw_channel').addEventListener('click', () => {
  const isDrawingActive = DRAWING_CONTROLL.get('active')

  if (isDrawingActive) {
    DRAWING_CONTROLL.set('active', false);
    return
  }

  DRAWING_CONTROLL.setAll({
    tool: "Parallel Channel",
    active: true,
  })
})
```

Is there any way I can change this behavior? Maybe I can catch the drawing completion event and set this value again or some other way?

**Comment by martynasma** [comment]
`drawingSelectionEnabled` is a setting for `StockChart`, not `DrawingControl`.

You need to set it there.

That said, the new logic is that "selection mode" resets back to `false` when you start drawing something, so that selection does not interfere with the drawing process.

We will discuss if there's maybe some place for a fuzzy logic there, like reverting back to selection mode after individual drawing is finished or when you exit drawing mode.

**Comment by OstapchelaEst** [comment]
@martynasma Thank you for your response! So there is currently no way to return this property to `true` when the drawing is finished?

**Comment by martynasma** [comment]
An event maybe:

```JavaScript
stockChart.events.on("drawingadded", function() {
  stockChart.set("drawingSelectionEnabled", true);
});
```

**Comment by martynasma** [comment]
We're changing the behavior in the next version: it will now restore "selection mode" automatically (provided it was on) after drawing is finished.

I'll let you know here when the new version is out.

Make sure you remove the [above event](https://github.com/amcharts/amcharts5/issues/1460#issuecomment-2057220764) when upgraded.

**Comment by martynasma** [comment]
Fixed in [5.9.2](https://github.com/amcharts/amcharts5/releases/tag/5.9.2).

## [5.9.2] - 2024-04-17

### Added
- Accessibility: Two new `Tooltip` settings added: `readerAnnounce` (boolean, default: `false`) and `labelAriaLabel`. If `readerAnnounce` is set to `true`, tooltip's contents (`labelAriaLabel` if set, or regular label text) will be read out by screen readers when tooltip is shown or its data item changes.
- Accessibility: New `Label` method: `getAccessibleText()`. Returns populated `arialLabel` text if set, or regular `text` content.
- New `Sprite` method: `compositeScale()`. Returns actual scale of an element, compounded based on scale of its own and all of its ancestors.
- Net global function `am5.getRootById(id)`. Returns `Root` instance stored in a `<div>` with specific id.
- 4 new `StockPanel` events added: `moved`, `closed`, `collapsed`, and `expanded`. [More details](https://www.amcharts.com/docs/v5/charts/stock/panels/#Events).
- 2 new `Entity` methods added: `off(key, callback?)` and `offPrivate(key, callback?)`. Allows removing setting/private setting value change events set via `on()`/`onPrivate()`. [More details](https://www.amcharts.com/docs/v5/concepts/events/#Removing).
- New `StockChart` setting: `erasingEnabled`. If set to `true`, the chart will go into "eraser" mode - same as clicking on an Eraser tool in stock toolbar.

### Changed
- HTML content will now scale according to its "composite" `scale`. (Scale calculated all the way its ancestor tree).
- `StockChart` will now restore "selection mode" after briefly turning it off while drawing is being drawn.
- Improved dangling of circular labels with Arabic text.

### Fixed
- HTML content was sometimes being incorrectly positioned if either `width` or `height` was set, but not both.
- Selecting a drawing on a `StockChart` was resulting in an error if there was no `DrawingControl` present in a chart's toolbar.
- `StockChart`'s eraser tool was not working properly since `5.9.1`.
- If a `StockPanel` was added to a zoomed `StockChart` it would not sync its zoom and would appear fully zoomed out.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
